File: AlgoExpert/Strings/Easy/Run Length Encoding.java
import java.util.*;

class Program {
  public String runLengthEncoding(String string) {
    // Write your code here.
		
		char currentChar = string.charAt(0);
		int counter = 0;
		StringBuilder sb = new StringBuilder();
		for(int i = 0; i < string.length(); i++){
			if(currentChar != string.charAt(i) || counter == 9){
				sb.append(counter).append(currentChar);
				counter = 0;
				currentChar = string.charAt(i);
			}
			counter++;
		}
		
		sb.append(counter).append(currentChar);
		
    return sb.toString();
  }
}

File: Python/AlgoExpert/Strings/Easy/runLengthEncoding.py
def runLengthEncoding(string):
    # Write your code here.
    curr_char = string[0]
    count = 0
    char_list = []

    for c in string:
        if c == curr_char:
            if count == 9:
                char_list.append(str(count))
                char_list.append(curr_char)
                count = 1
            else:
                count += 1
        else:
            char_list.append(str(count))
            char_list.append(curr_char)
            count = 1
            curr_char = c

    if count != 0:
        char_list.append(str(count))
        char_list.append(curr_char)

    return ''.join(char_list)
